### PR TITLE
Add function to erase whole nvs partition

### PR DESCRIPTION
**Pull request by schurwanzn-stabl**
Add safe wrapper around `nvs_flash_erase` and `nvs_flash_erase_partition` (https://docs.espressif.com/projects/esp-idf/en/v5.0.2/esp32/api-reference/storage/nvs_flash.html#_CPPv415nvs_flash_erasev) to wipe config partition completely.

### CODE REVIEW

File: src/nvs.rs [hunk 1]
```diff
@@ -110,6 +122,19 @@ impl NvsCustom {
 
         Ok(Self(c_partition))
     }
+
+    fn erase(partition: &str) -> Result<(), EspError> {
+        let mut registrations = NONDEFAULT_LOCKED.lock();
+        let c_partition = CString::new(partition).unwrap();
```
Comment (stabl-gjn) [inline src/nvs.rs]:
```suggestion
        let c_partition = CString::new(partition).unwrap();
        let mut registrations = NONDEFAULT_LOCKED.lock();
```

Could switch the lines to reduce time with locked mutex.

File: src/nvs.rs [hunk 2]
```diff
@@ -110,6 +122,19 @@ impl NvsCustom {
...
+
+        if registrations.contains(c_partition.as_ref()) {
+            return Err(EspError::from(ESP_ERR_INVALID_STATE).unwrap());
+        }
```
Comment (stabl-gjn) [inline src/nvs.rs]: For my understanding: Why is it a invalid state if the registrations contain the partition name?

Maybe also add this as a comment?

**Review by stabl-gjn — APPROVED** [review]
LGTM. Just a really minor thing. Feel free to merge.

File: src/nvs.rs (hunk 2 above)
Comment (stabl-gjn) [reply src/nvs.rs]: Comments helped. Thanks :)

**Comment by stabl-gjn** [comment]
Feel free to merge

File: src/nvs.rs (hunk 2 above)
Comment (schurwanzn-stabl) [reply src/nvs.rs]: `esp-idf-svc` doesn't seem to have any proper error reporting for errors originating in it's own code. instead a generic ESP-IDF invalid state error is returned. for what this error means see the now added comments :) good point

File: src/nvs.rs (hunk 1 above)
Comment (schurwanzn-stabl) [reply src/nvs.rs]: actually led me to release the lock while erasing since I noticed it might take some time on bigger partitions

**Pull request merged**